homeassistant/components/airscape/fan.py

```diff
@@ -0,0 +1,173 @@
+"""Platform for Airscape fan integration."""
+import logging
+import airscape
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+# Import the device class from the component that you want to support
+from homeassistant.components.fan import (
+    SUPPORT_SET_SPEED,
+    PLATFORM_SCHEMA,
+    FanEntity,
+    DOMAIN,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TIMEOUT, CONF_MINIMUM
+
+DEFAULT_TIMEOUT = 5
+DEFAULT_MINIMUM = 1
+
+_LOGGER = logging.getLogger(__name__)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_NAME): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_MINIMUM, default=DEFAULT_MINIMUM): cv.positive_int,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Airscape Fan platform."""
+    # Setup connection to the fan
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    minimum = config.get(CONF_MINIMUM)
+    timeout = config.get(CONF_TIMEOUT)
+
+    try:
+        device = airscape.Fan(host, timeout)
+    except (airscape.exceptions.Timeout, airscape.exceptions.ConnectionError):
+        _LOGGER.error(
+            "Cannot connect to %s.  " "Device did not respond to API request", host
+        )
+    else:
+        # Add devices
+        add_entities([AirscapeWHF(device, name, minimum)], True)
+
+    def service_speed_up(call):
+        """Handle speed_up service call."""
+        entity_id = call.data.get("entity_id")
+        _LOGGER.debug("Calling speed_up for %s", entity_id)
+
+        entity = hass.data[DOMAIN].get_entity(entity_id)
+        entity.speed_up()
+
+    def service_slow_down(call):
+        """Handle slow_down service call."""
+        entity_id = call.data.get("entity_id")
+        _LOGGER.debug("Calling slow_down for %s", entity_id)
+
+        entity = hass.data[DOMAIN].get_entity(entity_id)
+        entity.slow_down()
+
+    hass.services.register(DOMAIN, "airscape_speed_up", service_speed_up)
+    hass.services.register(DOMAIN, "airscape_slow_down", service_slow_down)
+
+    return True
+
+
+class AirscapeWHF(FanEntity):
+    """Representation of an Airscape Fan."""
+
+    def __init__(self, device, name, minimum):
+        """Initialize a AirscapeFan."""
+        self._fan = device
+        self._name = name
+        self._state = None
+        self._speed = None
+        self._available = True
+        self._minimum_speed = minimum
+        self._speed_list = [f"{i}" for i in range(0, 11)]
```

Comment:
Only speeds available in the base fan component are allowed in the speed list. Try to map as best possible between those and the device speed settings.

Reply:
https://github.com/home-assistant/architecture/issues/127

Reply:
Do you want to wait with continuing here until the architecture issue is resolved? If so please close this PR while waiting.

Reply:
I was hoping to restart the conversation in the architecture repo.  But it doesn't seem like anyone has the bandwidth to deal with this right now.  I guess the best thing to do is close this and just move my platform over to a custom component until the architecture issue gets resolved.